frontend/src/store.ts

```diff
@@ -0,0 +1,48 @@
+import { configureStore, createSlice, PayloadAction } from '@reduxjs/toolkit';
+
+interface State {
+  isApiKeyModalOpen: boolean;
+  apiKey: string;
+  isMenuOpen: boolean;
+}
+
+const initialState: State = {
+  isApiKeyModalOpen: false,
+  apiKey: '',
+  isMenuOpen: false,
+};
+
+export const slice = createSlice({
+  name: 'app',
+  initialState,
+  reducers: {
+    toggleApiKeyModal: (state) => {
+      state.isApiKeyModalOpen = !state.isApiKeyModalOpen;
+      console.log('showApiKeyModal', state.isApiKeyModalOpen);
+    },
+    setApiKey: (state, action: PayloadAction<string>) => {
+      state.apiKey = action.payload;
+      console.log('setApiKey', action.payload);
+    },
+    toggleIsMenuOpen: (state) => {
+      state.isMenuOpen = !state.isMenuOpen;
+    },
+  },
+});

```

Comment:
I see a single slice for entire app at the moment, which is fine for now. As we move ahead and things get more complex, we can split our redux store.

Reply:
Yeah just started learning redux today and just wanted to get something working. Will likely make multiple slices later